Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -725,74 +725,63 @@ namespace AzToolsFramework
-                }
-
-                AZ_Assert(priorColumnIndex != -1, "Tried to share column with an out of bounds index!");
-                if (priorColumnIndex != -1)
-                {
-                    m_columnLayout->AddSharePriorColumn(priorColumnIndex, domIndex);
-                }
+                priorColumnIndex = m_columnLayout->indexOf(m_domOrderedChildren[searchIndex]);
             }
-            else
+
+            AZ_Assert(priorColumnIndex != -1, "Tried to share column with an out of bounds index!");
+            if (priorColumnIndex != -1)
             {
-                m_columnLayout->RemoveSharePriorColumn(domIndex);
+                m_columnLayout->AddSharePriorColumn(priorColumnIndex, domIndex);
             }
         }
-
-        if (updatedInfo.m_descriptionString != currentInfo.m_descriptionString)
+        else
         {
-            setToolTip(
-                QString::fromUtf8(updatedInfo.m_descriptionString.data(), aznumeric_cast<int>(updatedInfo.m_descriptionString.size())));
-            childWidget->setToolTip(
-                QString::fromUtf8(updatedInfo.m_descriptionString.data(), aznumeric_cast<int>(updatedInfo.m_descriptionString.size())));
+            m_columnLayout->RemoveSharePriorColumn(domIndex);
         }
 
-        if (updatedInfo.m_isDisabled != currentInfo.m_isDisabled)
+        // Remove any cached attribute info that is default, else cache it for the layout to use.
+        // This way we only cache what is needed.
+        auto layoutAttributeInfoIter = m_childIndexToCachedAttributeInfo.find(domIndex);
+        if (layoutAttributeInfoIter != m_childIndexToCachedAttributeInfo.end() && updatedLayoutAttributes.IsDefault())
         {
-            childWidget->setEnabled(!updatedInfo.m_isDisabled);
+            m_childIndexToCachedAttributeInfo.erase(layoutAttributeInfoIter);
         }
-
-        if (attributeIter != m_childIndexToAttributeInfo.end() && updatedInfo.IsDefault())
+        else
         {
-            m_childIndexToAttributeInfo.erase(attributeIter);
+            m_childIndexToCachedAttributeInfo[domIndex] = updatedLayoutAttributes;
         }
-        else
+
+        AZStd::string_view descriptionView = AZ::Dpe::Nodes::PropertyEditor::Description.ExtractFromDomNode(domArray).value_or("");
+        QString descriptionString = QString::fromUtf8(descriptionView.data(), aznumeric_cast<int>(descriptionView.size()));
+        if (descriptionString != childWidget->toolTip())
         {
-            m_childIndexToAttributeInfo[domIndex] = updatedInfo;
+            setToolTip(descriptionString);
+            childWidget->setToolTip(descriptionString);
         }
+
+        bool isDisabled = AZ::Dpe::Nodes::PropertyEditor::Disabled.ExtractFromDomNode(domArray).value_or(false) ||
+            AZ::Dpe::Nodes::PropertyEditor::AncestorDisabled.ExtractFromDomNode(domArray).value_or(false);
+        childWidget->setEnabled(!isDisabled);
```

Comment:
Does this mean that for these two attributes they are always set by what we see from the node?

Reply:
For the Disabled attribute, that's correct. We can just set it directly each time because Qt is already checking for changes before performing the enabling/disabling stuff it does:
```
if (enable != q->testAttribute(Qt::WA_Disabled))
    return; // nothing to do
```

For the tooltip string, we don't need to store it twice (caching it) since we can just verify that the current tooltip string is different. There is some tradeoff since the DOM node stores the string as a string_view, so we have to convert it in order to compare with the tooltip.

The bug being fixed here happened because both the attribute cache and the DOM node said the widget was disabled, but the QWidget was actually enabled. So we weren't disabling the widget because our cache said that it already was (I think this used to be true). Same thing could happen with the tooltip; maybe with anything widget related.

Reply:
Thanks for the explanation. It is more clear to me now.